src/components/datasets/DatasetDataTypes.js

```diff
@@ -0,0 +1,128 @@
+    const dispatch = useDispatch();
+
+    const datasetDataTypes = useSelector((state) => state.datasetDataTypes.itemsById[dataset.identifier]);
+    const datasetSummaries = useSelector((state) => state.datasetSummaries.itemsById[dataset.identifier]);
+
+    const [datatypeSummaryVisible, setDatatypeSummaryVisible] = useState(false);
+    const [selectedDataType, setSelectedDataType] = useState(null);
+
+    const selectedSummary = (selectedDataType !== null && datasetSummaries)
+        ? datasetSummaries[selectedDataType.id]
+        : {};
+
+    const handleDeleteDataType = async (dataType) => {
+        genericConfirm({
+            title: `Are you sure you want to delete the "${dataType.label || ""}" data type?`,
+            content: "Deleting this means all instances of this data type contained in the dataset " +
+                "will be deleted permanently, and will no longer be available for exploration.",
+            onOk: async () => {
+                await dispatch(clearDatasetDataType(dataset.identifier, dataType.id));
+                await dispatch(fetchDatasetDataTypesSummaryIfPossible(dataset.identifier));
+            },
+        });
+    };
+
+    const showDatatypeSummary = (dataType) => {
+        setSelectedDataType(dataType);
+        setDatatypeSummaryVisible(true);
+    };
+
+    const dataTypesColumns = [
+        {
+            title: "Name",
+            key: "label",
+            render: (dt) =>
+                isPrivate ? (
+                    <a style={{ fontFamily: "monospace" }} onClick={() => showDatatypeSummary(dt)}>
+                        {dt.label ?? NA_TEXT}
+                    </a>
+                ) : dt.label ?? NA_TEXT,
+            defaultSortOrder: "ascend",
+            sorter: (a, b) => a.label.localeCompare(b.label),
+        },
+        {
+            title: "Count",
+            dataIndex: "count",
+            render: (c) => (c ?? NA_TEXT),
+        },
+        ...(isPrivate ? [
+            {
+                title: "Actions",
+                key: "actions",
+                width: 230,
+                render: (dt) => (
+                    <Row gutter={10}>
+                        <Col span={12}>
+                            <Button
+                                icon="import"
+                                style={{ width: "100%" }}
+                                onClick={() => (onDatasetIngest || nop)(project, dataset, dt)}
+                            >
+                                Ingest
+                            </Button>
+                        </Col>
+                        <Col span={12}>
+                            <Button
+                                type="danger"
+                                icon="delete"
+                                onClick={() => handleDeleteDataType(dt)}
+                                style={{ width: "100%" }}
+                            >
+                                Delete
```

Comment:
this should not be `Delete`, it should be `Clear Data` - since the data type stays, just the data of that type are removed.